### DISCUSSION TITLE
Bluetooth Controller

### DISCUSSION THREAD

**Discussion by WhoIsMrSentry** [makerforgetech/modular-biped#52] [Ideas]
Hi, I designed the bluetooth controller wristband that came to my mind for the robot and wrote the codes. includes 1pcs node32s  1pcs 16x2 screen and 5pcs potantiometer)

I made bluetooh mode as an alternative to live mode, keyboard mode and gamepad mode. Example that will be worn on the wrist like a watch and will be a certain set of combinations.
            
- k1=between 4080-4096     
- k2=0     
- k3=0     
- k4=0      
- k5=0

As for the values, it will write these values ​​to the robot via bluetooth, determine which function or animation to trigger from the combination sequence on the robot and start that function, as an example sit animation.

**footnote**: potantiometers gives 4096 value in node32s originally 1023 value in arduino

![top](https://github.com/dmt-labs/modular-biped/assets/134494889/1b9de37e-58ce-432a-882e-c4b3c0fca95b)
![right](https://github.com/dmt-labs/modular-biped/assets/134494889/70b43252-ef87-4af2-ba61-e0951082b8ae)

    <code>

        #include <BluetoothSerial.h>

        BluetoothSerial SerialBT;

        void setup() {
          SerialBT.begin("SentryESP32");  // Set the Bluetooth device name here
          Serial.begin(115200);  // Serial communication baud rate

          // Set GPIO pins for potentiometers
          pinMode(34, INPUT);
          pinMode(35, INPUT);
          pinMode(32, INPUT);
          pinMode(33, INPUT);
          pinMode(25, INPUT);
        }

        void loop() {
          // Read potentiometer values
          int potValue1 = analogRead(34);
          int potValue2 = analogRead(35);
          int potValue3 = analogRead(32);
          int potValue4 = analogRead(33);
          int potValue5 = analogRead(25);

          // Print potentiometer values in a single line
          Serial.printf("Thumb k1: %d, k2: %d, k3: %d, k4: %d, k5: %d\n", potValue1, potValue2, potValue3, potValue4, potValue5);

          // If a device is connected, print the device name and potentiometer values
          if (SerialBT.hasClient()) {
            SerialBT.printf("Thumb k1: %d, k2: %d, k3: %d, k4: %d, k5: %d\n", potValue1, potValue2, potValue3, potValue4, potValue5);
          }

          // Add some delay if needed
          delay(200);
        }

    </code>

**Comment by danic85** [comment]
Great idea @SentryCoderDev! do you have a schematic for the electronics? I'm interested to see how it is assembled.

**Reply by WhoIsMrSentry** [reply]
@danic85 Yes, I created a repository for this idea, everything is there.
[Node32s-BT_Wristband](https://github.com/SentryCoderDev/Node32s-BT_Wristband)
